task/sast-shell-check/0.1/sast-shell-check.yaml

```diff
@@ -0,0 +1,184 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: sast-shell-check
+spec:
+  description: >-
+    ShellCheck is a static analysis tool, gives warnings and suggestions for bash/sh shell scripts.
+  results:
+    - description: Tekton task test output.
+      name: TEST_OUTPUT
+  params:
+    - name: image-url
+      description: Image URL.
+      type: string
+      default: ""
+    - name: image-digest
+      description: Image digest to report findings for.
+      type: string
+      default: ""
+    - name: KFP_GIT_URL
+      description: git repository to download known false positives files from
+      type: string
+      # FIXME: Red Hat internal projects will default to https://gitlab.cee.redhat.com/osh/known-false-positives.git
+      # when KONFLUX-4530 is resolved
+      default: ""
+    - name: PROJECT_NVR
+      description: Name-Version-Release (NVR) of the scanned project, used to find path exclusions
+      type: string
+      default: ""
+    - name: RECORD_EXCLUDED
+      type: string
+      description: |
+        Whether to record the excluded findings (default to false).
+        If `true`, the excluded findings will be stored in `excluded-findings.json`.
+      default: "false"
+    - name: IMP_FINDINGS_ONLY
+      type: string
+      description: Whether to include important findings only
+      default: "true"
+  steps:
+    - name: sast-shell-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.7@sha256:cf6808a3bd605630a5d9f20595ff7c43f8645c00381219d32f5a11e88fe37072
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      env:
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+      script: |
+        #!/usr/bin/env bash
+        set -x
```

Comment:
We are printing all logs coming from every command. Do we want to have this?
In previous comments I received, they stated that we should not print everything to only provide useful comments to the users.

Reply:
I dare to say Yes.  Unless this produces too much noise on glob expansion, loop iteration, etc., the output of `set -x` (a.k.a XTRACE) usually provides the most useful data for debugging because it shows us the real commands that were executed.  For example, `rpmbuild` also enables XTRACE by default in the generated build scripts.  The problem with printing the fancy looking messages such as `executing Snyk Client ... done` instead of XTRACE is that:

- They hide important details.
- They may become incorrect over time because the code that prints these messages needs to by manually kept in sync with other code changes (unlike XTRACE).

I think this is a matter of taste.  Different users/developers will have different opinion on this.